### PR TITLE
Renames tasks directory to Tasks during upgrade

### LINKED ISSUE #7932: Upgrader does not fix $tasksdir
#### Description
Just a reference from #7930.  When tasks was renamed to 'Tasks' to fix other issues and align to more standards, the underlaying setting is not updated in Settings.php.  Upgrader needs to resolve this.
Setting this to Alpha 2 as we don't have the new upgrader logic, but it needs tracked so we fix it.

### PR DESCRIPTION
**Pull request by Sesquipedalian**
Fixes #7932

### CODE REVIEW

File: other/upgrade.php [hunk 1]
```diff
@@ -1839,6 +1839,13 @@ function DeleteUpgrade()
 		'upgradeData' => null,
 	);
 
+	// Fix case of Tasks directory.
+	if (is_dir(Config::$tasksdir) && is_writable(Config::$tasksdir) && is_writable(dirname(Config::$tasksdir))) {
+		rename(Config::$tasksdir, dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks_temp');
+		rename(dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks_temp', dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks');
+		$changes['tasksdir'] = dirname(Config::$tasksdir) . '/Tasks';
```
Comment (jdarwood007) [inline other/upgrade.php:1869]: I assume this should be $sourcedir?

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: In 99.999% of cases, `dirname(Config::$tasksdir)` will be the same as `Config::$sourcedir`, yes. But in theory someone might have moved it, so it's best to use `dirname(Config::$tasksdir)` to be sure.

Comment (jdarwood007) [reply other/upgrade.php:1869]: I missed the dirname() call.

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: Ah. Got it. Easy enough to overlook. 🙂

Comment (jdarwood007) [reply other/upgrade.php:1869]: Only thing I would suggest is maybe we check if we need to rename it or not.  I wonder if realpath check would suffice.  `realpath($taskdir) != realpath(dirname($taskdir) . '/Tasks')`?

Comment (jdarwood007) [reply other/upgrade.php:1869]: Also, is_dir('tasks') will not be true on case sensitive systems.  Is this intended to handle that case?

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: Good idea.

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: > Also, is_dir('tasks') will not be true on case sensitive systems. Is this intended to handle that case?

If `./Sources/tasks` has already been renamed to `./Sources/Tasks`, then getting false from that check is exactly what we want.

Comment (jdarwood007) [reply other/upgrade.php:1869]: Well not if `$tasksdir` is set to `$sorucedir . '/tasks' and we have now uploaded it as 'Tasks'.  Resulting in on case sensitive OS having both a 'tasks' and 'Tasks' directory.  Happened on one of my test boards when I manually updated it.

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: Ah. Yes, now I see what you mean. Hm... that does pose a bit of a problem, doesn't it? Got any ideas?

Comment (jdarwood007) [reply other/upgrade.php:1869]: Not sure yet.  Windows doesn't care, it sees it as the same.  Linux will see this as if they where named 2 different things.  PHP we can use strcasecmp to check them.  Maybe `strcasecmp($taskdir) != strcasecmp(dirname($taskdir) . '/Tasks');`?

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: I think I figured out a way to consistently handle all cases.

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: Oh, wait, nope. I thought realpath() would report the true case of the dir's name on case insensitive file systems. Apparently that was wrong, which means I need to think about this further. Ugh.

Comment (Sesquipedalian) [reply other/upgrade.php:1869]: Got it now. This is exactly what inodes are for.

**Comment by jdarwood007** [comment]
Honestly, just thinking on this.  Do we really need to specify `$tasksdir`?  Can we just eliminate this as a option for 3.0?  Tasks will always be `$sourcedir . '/Tasks'`.  The entire issue goes away completely.  To add to this, why don't we make Actions directory a config option, Unicode? Cache? Db? Graphics?  Where does it end.

**Comment by Sesquipedalian** [comment]
That's not a bad argument. But in that case, we need to add logic to the upgrader to move a custom tasks dir back to the standard location.

**Comment by Sesquipedalian** [comment]
At any rate, such a change would be Alpha 2, not Alpha 1.

**Comment by Sesquipedalian** [comment]
Ugh, I see this got closed. Apparently that happens automatically if you rename the branch that you want to merge. I'll open a new PR.

**Pull request closed without merging**